File: projects/08/Java/TemperatureTester.java
/* TemperatureTester.java tests the module/aggregate type Temperature
 * cs 214
 * Calvin College
 * Bryce Allen
 * 4/6/18
 */
import java.util.Scanner;
public class TemperatureTester
{
    public static void main(String[] args)
    {
	Temperature baseTemp = new Temperature(0.0, 'F');
	Temperature limitTemp = new Temperature(100.0, 'C');
	double stepValue = 2.5;

	while( !(baseTemp.equals(limitTemp)) )
        {
	    baseTemp.toCelsius(baseTemp.getDegrees());
	    System.out.print(baseTemp.printTemperature() + " ");
	    baseTemp.toKelvin(baseTemp.getDegrees());
	    System.out.print(baseTemp.printTemperature() + " ");
	    baseTemp.toFahrenheit(baseTemp.getDegrees());
	    System.out.print(baseTemp.printTemperature() + " ");
	    baseTemp = baseTemp.raise(baseTemp, stepValue);
	    System.out.println("");
	}

       	Temperature baseTemp1 = new Temperature(0.0, 'K');
	Temperature limitTemp1 = new Temperature(212.0, 'F');
	double stepValue1 = 20.0;

	while( !(baseTemp1.equals(limitTemp)) )
        {
	    baseTemp1.toCelsius(baseTemp1.getDegrees());
	    System.out.print(baseTemp1.printTemperature() + " ");
	    baseTemp1.toKelvin(baseTemp1.getDegrees());
	    System.out.print(baseTemp1.printTemperature() + " ");
	    baseTemp1.toFahrenheit(baseTemp1.getDegrees());
	    System.out.print(baseTemp1.printTemperature() + " ");
	    baseTemp1 = baseTemp1.raise(baseTemp1, stepValue1);
	    System.out.println("");
	}
    }
}

File: projects/08/Ruby/temperatureTester.rb

require 'test/unit/assertions'
require "./Temperature.rb"
include Test::Unit::Assertions

def testTemperature
  baseTemp = Temperatures::Temperature.new(0.0, "F")
  limitTemp = Temperatures::Temperature.new(100.0, "C")
  stepValue = 2.5
  
  while baseTemp.lessThan(limitTemp) == true
    #baseTemp.printTemp
    baseTemp.getCelsius(baseTemp)
    baseTemp.printTemp
    baseTemp.getKelvin(baseTemp)
    baseTemp.printTemp
    baseTemp.getFahrenheit(baseTemp)
    baseTemp.printTemp
    baseTemp = baseTemp.raise(baseTemp, stepValue)
    print "\n"
  end

  baseTemp1 = Temperatures::Temperature.new(0.0, "K")
  limitTemp1 = Temperatures::Temperature.new(212.0, "F")
  stepValue1 = 2.5
  
  while baseTemp1.lessThan(limitTemp1) == true
    #baseTemp.printTemp
    baseTemp1.getCelsius(baseTemp1)
    baseTemp1.printTemp
    baseTemp1.getKelvin(baseTemp1)
    baseTemp1.printTemp
    baseTemp1.getFahrenheit(baseTemp1)
    baseTemp1.printTemp
    baseTemp1 = baseTemp1.raise(baseTemp1, stepValue1)
    print "\n"
  end
end

testTemperature
